src/bootstrap/healthcheck.js

```diff
@@ -0,0 +1,96 @@
+import network from "../store/selectors/network";
+import {
+  setLastNetworkCheckEndAt,
+  setLastNetworkCheckStartAt
+} from "../store/reducers/network/onNetworkCheckEndEpic";
+import platformReducer from "../store/reducers/platform";
+import networkReducer from "../store/reducers/network";
+import * as Network from "./network";
+import { checkIfOutOfSyncEpic } from "../store/reducers/network/checkIfOutOfSync";
+import Raven from "raven-js";
+import { registerSubscription } from "../utils/subscriptions/registerSubscription";
+import {
+  HAS_ACCOUNTS,
+  SUBSCRIPTIONS_GROUP_GLOBAL_INITIAL,
+  SUBSCRIPTIONS_LATEST_BLOCK
+} from "../constants";
+import { subscribeLatestBlockFilterEpic } from "../store/reducers/network/subscribeLatestBlockFilterEpic";
+import accounts from "../store/selectors/accounts";
+import accountsReducer from "../store/reducers/accounts";
+import { Session } from "../utils/session";
+import { CheckNetworkAction } from "../store/reducers/network/CheckNetworkAction";
+
+export const healthCheck = async (
+  dispatch,
+  getState,
+  isInitialHealthcheck = false
+) => {
+  if (network.isNetworkCheckPending(getState()) === true) {
+    return;
+  }
+  dispatch(setLastNetworkCheckStartAt());
+  if (isInitialHealthcheck) {
+    dispatch(platformReducer.actions.setGlobalFormLockEnabled());
+    dispatch(networkReducer.actions.connecting());
+  }
+  try {
+    const providerType = await Network.checkConnectivity().catch();
+    const connectedNetworkId = await dispatch(
+      networkReducer.actions.getConnectedNetworkId()
+    );
+    if (isInitialHealthcheck) {
+      await dispatch(checkIfOutOfSyncEpic());
+      // console.log("connectedTo:", network.activeNetworkMeta(getState()).get('name'));
+      Raven.setTagsContext({
+        network: network.activeNetworkMeta(getState()).get("name")
+      });
+    }
+    if (providerType && connectedNetworkId.value) {
+      dispatch(networkReducer.actions.connected());
+      if (isInitialHealthcheck) {
+        /**
+         * We only do this once, since later we subscribe to 'latest' filter
+         * to get notified on new block resolved
+         */
+        dispatch(networkReducer.actions.getLatestBlockNumber());
```

Comment:
Out of curiosity. If we have a js file  that wants something from the store, can't we juts inject the store and get the what we are looking for instead of creating a  get method for  everything?